### PR TITLE
Non-aux gun mode gunmod fix

### PR DESCRIPTION
**Pull request by tenmillimaster**
This should fix gunmods that add  Non-auxiliary gun modes, while not breaking bayonets.
#### Summary

  ```SUMMARY: Bugfixes "fixes gunmods that add  Non-auxiliary gun modes "```

#### Purpose of change
Back in ```#20980```, the autofire mechanism was reported broken and was  removed with ```#209534``` . I was hoping to add in some similar gunmods, but found they were still bugged.
Hopefully this paves the way for more sensible gunmods, like an auto-sear (e.g. lightning links and RDIAS's for AR15's),  trigger disconnectors (e.g. the rifle trigger pack for a stoner 63, or every semiautomatic conversion ever), and burst counters (e.g. HK G3/ MP5 burst trigger packs and M16 trigger groups). Lightning links are literally just a piece of springsteel used to disengage the disconnector, such ~~impeccable~~ shoddy craftsmanship deserves a place in the apocalypse.

#### Describe the solution

Following the fix back in ```#19515```, I changed what item was under consideration, and added an if statement for firemodes that were "REACH" type so bayonets would be unaffected.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -5332,12 +5332,18 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
...
         } else if( e->is_gunmod() ) {
-            for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
+            for( auto m : e->type->gunmod->mode_modifier )
+                  if( m.first == ("REACH"))   {
+				res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
                                                  m.second.qty(), m.second.flags() } );
-            }
+            } else {
+        		res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( this ),
```
Comment (mlangsdorf) [inline src/item.cpp]: Needs two more spaces to be tabbed correctly.

File: src/item.cpp [hunk 2]
```diff
@@ -5332,12 +5332,18 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
...
-            // non-auxiliary gunmods may provide additional modes for the base item
+            // non-auxiliary gunmods may provide additional modes for the base item
+
+
         } else if( e->is_gunmod() ) {
-            for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
+            for( auto m : e->type->gunmod->mode_modifier )
+                  if( m.first == ("REACH"))   {
+				res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
```
Comment (mlangsdorf) [inline src/item.cpp]: Needs two more spaces in front to be tabbed correctly.

File: src/item.cpp [hunk 3]
```diff
@@ -5332,12 +5332,18 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
...
 
-            // non-auxiliary gunmods may provide additional modes for the base item
+            // non-auxiliary gunmods may provide additional modes for the base item
+
+
         } else if( e->is_gunmod() ) {
-            for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
+            for( auto m : e->type->gunmod->mode_modifier )
+                  if( m.first == ("REACH"))   {
```
Comment (mlangsdorf) [inline src/item.cpp]: The () around "REACH" are unnecessary, and each '(' or ')' should be separated from other characters by a space on each side.  Also, there should be only one space between the final ')' and the '{'.

**Comment by tenmillimaster** [comment]
I think everything is fixed now?

**Comment by mlangsdorf** [comment]
NACK.

Please don't astyle all of item.cpp.  We know it is poorly astyle'd.  Please only astyle your specific changes.

**Comment by tenmillimaster** [comment]
Mon dieu! Sorry, Let's try that again.

**Comment by tenmillimaster** [comment]
I think I un-did all the mess.

**Comment by tenmillimaster** [comment]
Severalth time's the charm? I added some comments to make it clearer what I did, just incase other non-reach melee gunmods ever show up.

File: src/item.cpp [hunk 4]
```diff
@@ -5333,17 +5333,25 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
             }
 
             // non-auxiliary gunmods may provide additional modes for the base item
-        } else if( e->is_gunmod() ) {
-            for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
-                                                 m.second.qty(), m.second.flags() } );
+        } else if( e->is_gunmod() )
+            {
```
Comment (Night-Pryanik) [inline src/item.cpp]: Raise `{` to the upper line like this:

`} else if( e->is_gunmod() ) {`

File: src/item.cpp [hunk 5]
```diff
@@ -5333,17 +5333,25 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
...
+                }
             }
-        }
     }
 
-    return res;
+return res;
 }
 
+
```
Comment (BevapDin) [inline src/item.cpp]: Random white space change (also a few lines above).

**Comment by tenmillimaster** [comment]
I've got a line of code to let gunmod jsons use item IDs as mod targets in addition to firearm types. Should that go here and should I edit the PR or should I open a new one? This is specifically so that json gunmods can easily target specific firearms instead of entire classes, This seemed necessary based off of the way gun mod design seems to be going.

File: src/item.cpp [hunk 6]
```diff
@@ -5334,14 +5334,19 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
 
             // non-auxiliary gunmods may provide additional modes for the base item
         } else if( e->is_gunmod() ) {
-            for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
-                                                 m.second.qty(), m.second.flags() } );
+            for( auto m : e->type->gunmod->mode_modifier )
+            //checks for melee gunmod, points to gunmod
+                if( m.first == "REACH") {
```
Comment (mlangsdorf) [inline src/item.cpp]:
```suggestion
                if( m.first == "REACH" ) {
```

File: src/item.cpp [hunk 7]
```diff
@@ -5334,14 +5334,19 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
...
+                } else {
+                    res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( this ),
+                                                     m.second.qty(), m.second.flags() } );
+                }
             }
-        }
     }
-
-    return res;
+return res;
```
Comment (mlangsdorf) [inline src/item.cpp]:
```suggestion
     return res;
```

File: src/item.cpp [hunk 8]
```diff
@@ -5334,14 +5334,19 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
...
+                if( m.first == "REACH") {
+                    res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
+                                                     m.second.qty(), m.second.flags() } );
+            //otherwise points to the parent gun, not the gunmod
+                } else {
+                    res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( this ),
+                                                     m.second.qty(), m.second.flags() } );
+                }
             }
-        }
```
Comment (mlangsdorf) [inline src/item.cpp]: please don't delete this brace or you're not closing off the for loop properly.

**Comment by mlangsdorf** [comment]
This is what you should be submitting.  Please submit this.
```diff
index dddcdc448f..e20a266bd3 100644
--- a/src/item.cpp
+++ b/src/item.cpp
@@ -5305,8 +5305,15 @@ std::map<gun_mode_id, gun_mode> item::gun_all_modes() const
             // non-auxiliary gunmods may provide additional modes for the base item
         } else if( e->is_gunmod() ) {
             for( auto m : e->type->gunmod->mode_modifier ) {
-                res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
-                                                 m.second.qty(), m.second.flags() } );
+                //checks for melee gunmod, points to gunmod
+                if( m.first == "REACH" ) {
+                    res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( e ),
+                                                     m.second.qty(), m.second.flags() } );
+                    //otherwise points to the parent gun, not the gunmod
+                } else {
+                    res.emplace( m.first, gun_mode { m.second.name(), const_cast<item *>( this ),
+                                                     m.second.qty(), m.second.flags() } );
+                }
             }
         }
     }
```

**Comment by kevingranade** [comment]
Jenkins rebuild

**Comment by tenmillimaster** [comment]
I've managed to do something completely wrong because github desktop wouldn't let me push without pulling. : /  Now there are TWO-HUNDRED and FORTY-ONE changed files.

At this point it'd be easier to just re-fork and insert the few lines again rather than clean this up. 

What a trainwreck. To say I am new to this would be an understatement.

**Comment by cainiaowu** [comment]
The trick I learn is never do revert in your PR branch. It is quite misleading since it dont actually revert but instead adding a new and totally opposite commit instead.

**Comment by mlangsdorf** [comment]
Don't refork.  Close this branch, open a new branch, apply my version of the commit, commit it, and push that new branch.

**Comment by tenmillimaster** [comment]
Moved to ````#27337````.

**Pull request closed without merging**